### ISSUE TITLE
Big requests error

### ISSUE THREAD

**Issue by GreenGood** [neo4j/neo4j-dotnet-driver#307]
When running the following code 
```
using (var session = _driver.Session())
            {
                await session.RunAsync(sb.ToString());
            }
```
with request from attached file i got a error
```
2018-05-31 19:21:33.6231|ERROR||Connection with the server breaks due to AggregateException: Произошла одна или несколько ошибок.
Neo4j.Driver.V1.ServiceUnavailableException: Connection with the server breaks due to AggregateException: Произошла одна или несколько ошибок.
   в Neo4j.Driver.Internal.Connector.PooledConnection.OnError(Exception error)
   в Neo4j.Driver.Internal.Connector.DelegatedConnection.OnErrorAsync(Exception error)
   в Neo4j.Driver.Internal.Connector.DelegatedConnection.<TaskWithErrorHandling>d__26.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в Neo4j.Driver.Internal.Result.ResultCursorBuilder.<>c__DisplayClass11_0.<<SetReceiveOneFunc>b__0>d.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в Neo4j.Driver.Internal.Result.ResultCursorBuilder.<GetKeys>d__8.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в Neo4j.Driver.Internal.Result.ResultCursorBuilder.<PreBuildAsync>d__7.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в Neo4j.Driver.Internal.Session.<>c__DisplayClass14_0.<<RunAsync>b__0>d.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в Neo4j.Driver.Internal.LoggerBase.<TryExecuteAsync>d__8`1.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   в CCS.RegistryExtract.Services.Implementation.Jobs.BuildGraphJob.<InsertNode>d__11.MoveNext() в C:\Work\Source\Fintender.CCS\CCS.RegistryExtract\Services\Implementation\Jobs\BuildGraphJob.cs:строка 234
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   в CCS.RegistryExtract.Services.Implementation.Jobs.BuildGraphJob.<Execute>d__8.MoveNext() в C:\Work\Source\Fintender.CCS\CCS.RegistryExtract\Services\Implementation\Jobs\BuildGraphJob.cs:строка 93
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   в CCS.RegistryExtract.Services.Implementation.Jobs.JobBase.<Execute>d__2.MoveNext() в C:\Work\Source\Fintender.CCS\CCS.RegistryExtract\Services\Implementation\Jobs\JobBase.cs:строка 38 --->
System.AggregateException: Произошла одна или несколько ошибок.
   в System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   в System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   в System.Threading.Tasks.Task`1.get_Result()
   в Neo4j.Driver.Internal.IO.BoltReader.<>c__DisplayClass11_0.<ReadAsync>b__0(Task`1 t)
   в System.Threading.Tasks.ContinuationTaskFromResultTask`1.InnerInvoke()
   в System.Threading.Tasks.Task.Execute()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в Neo4j.Driver.Internal.Connector.SocketClient.<ReceiveOneAsync>d__21.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в Neo4j.Driver.Internal.Connector.SocketClient.<ReceiveOneAsync>d__21.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в Neo4j.Driver.Internal.Connector.SocketConnection.<ReceiveOneAsync>d__21.MoveNext()
--- Конец трассировка стека из предыдущего расположения, где возникло исключение ---
   в System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   в System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   в Neo4j.Driver.Internal.Connector.DelegatedConnection.<TaskWithErrorHandling>d__26.MoveNext()
 --->
System.IO.IOException: Unexpected end of stream, read returned 0.
   в Neo4j.Driver.Internal.IO.ChunkReader.<>c__DisplayClass12_0.<ReadNextChunkLoopAsync>b__1(Task`1 t)
   в System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   в System.Threading.Tasks.Task.Execute()
```
Same request executes fine in browser
[request.txt](https://github.com/neo4j/neo4j-dotnet-driver/files/2059146/request.txt)

.Net Framework 4.7

**Comment by ali-ince** [comment]
Hi @GreenGood,

I've tried executing the statement you've provided on a 3.3.5 server and 1.6.1 driver without running into any kind of problems.

Could you please provide more information about your environment, including;

1. What version of database are you running?
2. What version of driver are you using?
3. Is the database you're running against is local or remote? If it is remote, can you please try with a local server?

Thanks.

**Comment by GreenGood** [comment]
1. 3.4.0 Community Edition
2. 1.6.1
3. I`am using local database.

After executing request against empty new database I`ve also found this in neo4j logs

```
2018-06-01 14:10:18.917+0000 ERROR [o.n.b.t.p.HouseKeeper] Fatal error occurred when handling a client connection: [id: 0x87007621, L:/127.0.0.1:7687 - R:/127.0.0.1:60369] Adjusted frame length exceeds 65535: 65537 - discarded
io.netty.handler.codec.TooLongFrameException: Adjusted frame length exceeds 65535: 65537 - discarded
	at io.netty.handler.codec.LengthFieldBasedFrameDecoder.fail(LengthFieldBasedFrameDecoder.java:522)
	at io.netty.handler.codec.LengthFieldBasedFrameDecoder.failIfNecessary(LengthFieldBasedFrameDecoder.java:500)
	at io.netty.handler.codec.LengthFieldBasedFrameDecoder.exceededFrameLength(LengthFieldBasedFrameDecoder.java:387)
	at io.netty.handler.codec.LengthFieldBasedFrameDecoder.decode(LengthFieldBasedFrameDecoder.java:430)
	at io.netty.handler.codec.LengthFieldBasedFrameDecoder.decode(LengthFieldBasedFrameDecoder.java:343)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:428)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1407)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1177)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1221)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:428)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by ali-ince** [comment]
Hi @GreenGood,

We've been able to reproduce the problem and created a server side fix [#11918](https://github.com/neo4j/neo4j/pull/11918) for that.

In the meantime, you can try to shorten your query text to get below 65535 bytes (maybe by removing redundant white-space) to get it running.

Thanks for reporting this issue.

**Comment by ali-ince** [comment]
Closing the issue as the fix is already part of the latest 3.4 releases.
